test/zone-spec/proxy.spec.ts

```diff
@@ -0,0 +1,120 @@
+import '../../lib/zone-spec/proxy';
+
+describe('ProxySpec', () => {
+  const ProxyZoneSpec: any = Zone['ProxyZoneSpec'];
+  const delegate = {name: 'delegate'};
+  const proxyZoneSpec = new ProxyZoneSpec(delegate);
+  const proxyZone = Zone.current.fork(proxyZoneSpec);
+
+  describe('properties', () => {
+    it('should expose ProxyZone in the properties', () => {
+      expect(proxyZone.get('ProxyZoneSpec')).toBe(proxyZoneSpec);
+    });
+
+    it('should assert that it is in or out of ProxyZone', () => {
+      expect(() => ProxyZoneSpec.assertPresent()).toThrow();
+      expect(ProxyZoneSpec.isLoaded()).toBe(false);
+      expect(ProxyZoneSpec.get()).toBe(undefined);
+      proxyZone.run(() => {
+        expect(ProxyZoneSpec.isLoaded()).toBe(true);
+        expect(() => ProxyZoneSpec.assertPresent()).not.toThrow();
+      expect(ProxyZoneSpec.get()).toBe(proxyZoneSpec);
+      });
+    });
+
+    it('should reset properties', () => {
+      expect(proxyZone.get('key')).toBe(undefined);
+      proxyZoneSpec.setDelegate({name: 'd1', properties: {'key': 'value'}});
+      expect(proxyZone.get('key')).toBe('value');
+      proxyZoneSpec.resetDelegate();
+      expect(proxyZone.get('key')).toBe(undefined);
+    });
+  });
+
+  describe('delegate', () => {
+    it('should set/reset delegate', () => {
+      const defaultDelegate: ZoneSpec = {name: 'defaultDelegate'};
+      const otherDelegate: ZoneSpec = {name: 'otherDelegate'};
+      const proxyZoneSpec = new ProxyZoneSpec(defaultDelegate);
+      const proxyZone = Zone.current.fork(proxyZoneSpec);
+
+      expect(proxyZoneSpec.getDelegate()).toEqual(defaultDelegate);
+
+      proxyZoneSpec.setDelegate(otherDelegate);
+      expect(proxyZoneSpec.getDelegate()).toEqual(otherDelegate);
+      proxyZoneSpec.resetDelegate();
+      expect(proxyZoneSpec.getDelegate()).toEqual(defaultDelegate);
+    }); 
+  });
+
+  describe('forwarding', () => {
+    const proxyZoneSpec = new ProxyZoneSpec();
```

Comment:
Can these be called in a `beforeEach`? When Jasmine calls `describe` can be confusing since all `describe` blocks will get called before any tests (I think...)